### ISSUE TITLE
Issue with $search query values containing whitespace

### ISSUE THREAD

**Issue by awright415** [jaystack/odata-v4-parser#10]
When I send a request to the server with a $search query that has whitespace, I get an "Unexpected Character" error, unless there's an AND/OR in between. So, in other words, the request fails on phrase queries and implicit AND queries. Examples:

- **Works:** parser.query("$search=foo AND bar")
- **Works:** parser.query("$search=\\"foo AND bar\\"")
- **Does not work:** parser.query("$search=foo bar")
- **Does not work:** parser.query("$search=\\"foo bar\\"")

**Comment by lazarv** [comment]
Hi @awright415,
the parser is based on the official ABNF and according to it, search phrase and word are:

```
searchPhrase = quotation-mark 1*qchar-no-AMP-DQUOTE quotation-mark
searchWord   = 1*ALPHA ; Actually: any character from the Unicode categories L or Nl,
                       ; but not the words AND, OR, and NOT
```

ABNF doesn't allow whitespace in search phrase and encoding needed to work as expected:

```
parser.query('$search="foo%20bar"')
```

This works as expected and parsed as a search phrase (because the quotation marks).

```
{
  position: 8,
  next: 19,
  value: 'foo%20bar',
  type: 'SearchPhrase',
  raw: '"foo%20bar"'
}
```

**Comment by awright415** [comment]
Thanks for the quick response! I got thrown off by [this section](http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/part1-protocol/odata-v4.0-errata03-os-part1-protocol-complete.html#_The_$search_System) of the protocol spec, which shows a phrase search example with unescaped whitespace.

Ultimately, the trouble I'm having is running $search queries with odata-v4-server, and perhaps I just submitted the wrong issue on the wrong project. I get 500 responses with the same unexpected character error for those requests (and naturally the whitespace is encoded in the URL).

It looks like inside of `processor.ts`, the URL is already unescaped before it's sent to the parser. So, in that case, wouldn't the parser be trying to parse a $search query value with unencoded whitespace like examples I provided?
